src/net.c

```diff
@@ -3238,10 +3292,21 @@ coap_join_mcast_group_intf(coap_context_t *ctx, const char *group_name,
 
         if (result == COAP_SOCKET_ERROR) {
           coap_log(LOG_ERR,
-                   "coap_join_mcast_group: setsockopt: %s: '%s'\n",
-                   coap_socket_strerror(), group_name);
+                   "coap_join_mcast_group_intf: %s: setsockopt: %s\n",
+                   group_name, coap_socket_strerror());
         }
         else {
+          char addr_str[INET6_ADDRSTRLEN + 8 + 1];
+
+          addr_str[sizeof(addr_str)-1] = '\000';
+          if (coap_print_addr(&gaddr, (uint8_t*)addr_str,
```

Comment:
`coap_print_addr()` does not guarantee that a terminating zero is written. A proper use would be something like the following:

```c
size_t written;
written = coap_print_addr(&gaddr, (uint8_t *)addr_str, sizeof(addr_str) - 1);
if (written > 0) {
  addr_str[written] = '\000';
  ...
}
```

Reply:
Interesting - all other callers of `coap_print_addr()` do not check for a terminating zero.  Making `coap_print_addr()` robust instead.

Reply:
Agreed